In [81]:
# importing requiring libraries
import pandas as pd
# importing plotly for visualization 
import plotly.express as px

In [82]:
df = pd.read_csv("/content/drive/MyDrive/Xethon/Retail Product Recommender.csv")

In [83]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 406829 entries, 0 to 406828
Data columns (total 7 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    406829 non-null  object 
 1   StockCode    406829 non-null  object 
 2   Description  406829 non-null  object 
 3   Quantity     406829 non-null  int64  
 4   InvoiceDate  406829 non-null  object 
 5   UnitPrice    406829 non-null  float64
 6   CustomerID   406829 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 21.7+ MB


No of Rows --- 406829


In [84]:
df.describe()


Quantity      UnitPrice     CustomerID
count  406829.000000  406829.000000  406829.000000
mean       12.061303       3.460471   15287.690570
std       248.693370      69.315162    1713.600303
min    -80995.000000       0.000000   12346.000000
25%         2.000000       1.250000   13953.000000
50%         5.000000       1.950000   15152.000000
75%        12.000000       3.750000   16791.000000
max     80995.000000   38970.000000   18287.000000

In [85]:
df.shape

(406829, 7)

In [86]:
df.head(10)

InvoiceNo StockCode  ... UnitPrice  CustomerID
0    536365    85123A  ...      2.55       17850
1    536365     71053  ...      3.39       17850
2    536365    84406B  ...      2.75       17850
3    536365    84029G  ...      3.39       17850
4    536365    84029E  ...      3.39       17850
5    536365     22752  ...      7.65       17850
6    536365     21730  ...      4.25       17850
7    536366     22633  ...      1.85       17850
8    536366     22632  ...      1.85       17850
9    536367     84879  ...      1.69       13047

[10 rows x 7 columns]

In [87]:
df.CustomerID.nunique()

4372

Findings no of 1 <br>
We have found that only **4372** Customers are active in purchasing regularly

In [88]:
df.StockCode.nunique()

3684

Findings no 2<br>
No of Unique Products --- 3684

In [89]:
df.InvoiceNo.nunique()

22190

In [98]:
a=df.groupby(["Description"],as_index=False).sum().drop(['UnitPrice','CustomerID'], axis = 1)

In [107]:
b=a.head(100)
b

Description  Quantity
0     4 PURPLE FLOCK DINNER CANDLES       140
1     50'S CHRISTMAS GIFT BAG LARGE      1883
2                 DOLLY GIRL BEAKER      2391
3       I LOVE LONDON MINI BACKPACK       360
4       I LOVE LONDON MINI RUCKSACK         1
..                              ...       ...
95    50'S CHRISTMAS PAPER GIFT BAG      1326
96  50CM METAL STRING WITH  7 CLIPS       222
97  6 CHOCOLATE LOVE HEART T-LIGHTS      1018
98         6 EGG HOUSE PAINTED WOOD       254
99      6 GIFT TAGS 50'S CHRISTMAS       3298

[100 rows x 2 columns]

In [92]:
print(type(b))

<class 'pandas.core.frame.DataFrame'>


In [93]:
!pip install plotly.express

Findings no 3 <br>
We have visualized products and their sale quantity. We can find that certain exclusive products have sold more than **5000 times**

In [109]:
plot_bar = px.data.gapminder()
px.bar(data_frame=b, x='Description',y='Quantity',color='Description')